systers_portal/community/forms.py

```diff
@@ -1,14 +1,123 @@
 from django import forms
+from django.forms import ValidationError
 from django.contrib.auth.models import Group
 
 from common.forms import ModelFormWithHelper
 from common.helpers import SubmitCancelFormHelper
-from community.constants import COMMUNITY_ADMIN
-from community.models import Community, CommunityPage
+from community.constants import COMMUNITY_ADMIN, COMMUNITY_PRESENCE_CHOICES
+from community.models import Community, CommunityPage, RequestCommunity
 from community.utils import get_groups
 from users.models import SystersUser
 
 
+class RequestCommunityForm(ModelFormWithHelper):
+    """Form to request a new Community"""
+    def __init__(self, *args, **kwargs):
+        """Makes some fields required and modifies a field to use widget"""
+        self.user = kwargs.pop('user')
+        super(RequestCommunityForm, self).__init__(*args, **kwargs)
+        self.fields['social_presence'] = forms.MultipleChoiceField(
+            choices=COMMUNITY_PRESENCE_CHOICES, label="Check off all \
+            the social media accounts you can manage for your proposed community:",
+            required=False, widget=forms.CheckboxSelectMultiple)
+        self.fields['email'].required = True
+        self.fields['demographic_target_count'].required = True
+        self.fields['purpose'].required = True
+        self.fields['content_developer'].required = True
+        self.fields['selection_criteria'].required = True
+        self.fields['is_real_time'].required = True
+
+    class Meta:
+        model = RequestCommunity
+        fields = ('is_member', 'email_id', 'email', 'name', 'slug', 'order',
+                  'type_community', 'other_community_type', 'parent_community',
+                  'community_channel', 'mailing_list', 'website', 'facebook',
+                  'googleplus', 'twitter', 'social_presence', 'other_account',
+                  'demographic_target_count',
+                  'purpose', 'is_avail_volunteer', 'count_avail_volunteer', 'content_developer',
+                  'selection_criteria', 'is_real_time')
+        helper_class = SubmitCancelFormHelper
+        helper_cancel_href = "{% url 'index' %}"
+
+    def clean_social_presence(self):
+        """Converts the checkbox input into char to save it to the instance's field."""
+        social_presence = ', '.join(
+            map(str, self.cleaned_data['social_presence']))
+        return social_presence
+
+    def save(self, commit=True):
+        """Override save to add user to the instance"""
+        instance = super(RequestCommunityForm, self).save(commit=False)
+        instance.user = SystersUser.objects.get(user=self.user)
+        if commit:
+            instance.save()
+        return instance
+
```

Comment:
Here as well the username is being passed and not a User instance, say I login as admin and create and submit a request, it would then suggest to have a User instance and not an admin instance, a `Value error at /community/name/`.
If I login as a user and create and submit a request it would then suggest me to have a User instance and not say test_user instance. The foreign key dependency needs to be handled here. Could you share what happens for you in these two cases?

Reply:
@tapasweni-pathak I took care of this issue. Updated my PR. Thanks.